projects/packages/my-jetpack/src/products/class-security.php

```diff
@@ -162,45 +160,13 @@ public static function is_active() {
 		return static::is_jetpack_plugin_active() && static::has_required_plan();
 	}
 
-	/**
-	 * Hits the wpcom api to check scan status.
-	 *
-	 * @todo Maybe add caching.
-	 *
-	 * @return Object|WP_Error
-	 */
-	private static function get_state_from_wpcom() {
-		static $status = null;
-
-		if ( $status !== null ) {
-			return $status;
-		}
-
-		$site_id = Jetpack_Options::get_option( 'id' );
-
-		$response = Client::wpcom_json_api_request_as_blog(
-			sprintf( '/sites/%d/purchases', $site_id ),
-			'1.1',
-			array(
-				'method' => 'GET',
-			)
-		);
-		if ( 200 !== wp_remote_retrieve_response_code( $response ) ) {
-			return new WP_Error( 'purchases_state_fetch_failed' );
-		}
-
-		$body   = wp_remote_retrieve_body( $response );
-		$status = json_decode( $body );
-		return $status;
-	}
-
```

Comment:
This is a good change but slightly outside the scope of the current PR, which is to initialize the stats product in My Jetpack.

Let's go ahead and ship this since `get_state_from_wpcom`'s and `get_site_current_purchases` are identical, but it'd be nice to split these refactors into a separate PR in the future.